vignettes/philosophy.Rmd

```diff
@@ -0,0 +1,163 @@
+  When `length(x) == 0`, `1:length(x)` yields `c(1, 0)`, whereas `seq_along(x)` would yield an empty vector.
+- Don't store code in comments.
+  If you don't want to loose code, use version control systems like [`git`](https://git-scm.com/).
+- Avoid code with lots of nested loop or if statements.
+  If the code it too complex, you'll get a warning that the [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) is too high.
+  Tips for reducing the code complexity:
+    - Use `assertthat::assert_that()` to validate object or conditions instead of `if() stop()`.
+    - See if you can use `ifelse()` instead of `if()`.
+    - Split the main function of your code over a few sub functions.
+
+## File name conventions
+
+To make this easier to remember we choose the same name conventions for file names as for objects.
+We acknowledge that these rules sometimes clash with requirements from other sources (e.g. `DESCRIPTION` in an R package, `README.md` on GitHub, `.gitignore` for git, ...).
+In such case we allow the file names as required by R, git or GitHub.
+When `check_filename()` does not allow a certain file or folder name and you think this is a false positive, then please open an [issue on GitHub](https://github.com/inbo/checklist/issues) and motivate why this is a false positive.
+
+### Rules for folder names
+
+- Folder names should only contain lower case letters, numbers and underscore (`_`).
+- They can start with a single dot (`.`).
+
+### Rules for file names
+
+- Base names should only contain lower case letters, numbers and underscore (`_`).
+- File extensions should only contains lower case letters and numbers.
+  Exceptions: file extensions related to R must have an upper case R (`.R`, `.Rmd`, `.Rd`, `.Rnw`, `.Rproj`).
+
+### Rules for graphical file names
+
+- Applies to files with extensions `csl`, `eps`, `jpg`, `jpeg`, `pdf`, `png` and `ps`.
+- Same rules except that you need to use a dash (`-`) as separator instead of an underscore (`_`).
+  We need to exception because underscores cause problems in certain situations.
+
+## Bundeling your code a package
+
+Most users think of an R package a collection of generic functions that they can use to run their analysis.
+However, an R package is useful way to bundle and document a stand alone analysis too!
+Suppose you want to pass your code to a collaborator or your future self who is working on a different computer.
+If you have a as project folder with a bunch of files, people will need to get to know your project structure, find out what scripts to run and which dependencies they need.
+Unless you documented everything well they (including your future self!) will a hard time figuring out how thing work.
+
+Having the analysis as a package _and_ running `check_package()` to ensure a minimal quality standard, make things a lot easier for the user.
+Agreed, it will take more time to create the analysis, especially with the first few projects.
```

Comment:
As it takes a lot of time, I certainly wouldn't make a package for every short project, I think it only has an advantage for long projects where analyses have to be done multiple times in subsequent years.  Or many small projects with similar analyses...  Maybe it is useful to add this element (repetition or longtime project) as well?